### ISSUE TITLE
Add support for `ignore` like gocyclo

### ISSUE THREAD

**Issue by wy98926** [uudashr/gocognit#18]
Shall we add support for the `ignore` in both command flag and code directive?

Example: [gocyclo](https://github.com/fzipp/gocyclo#ignoring-individual-functions)

We expect to do as the following:

- Command flag

	```bash
	$ gocognit
	Usage:
			gocognit [flags] <Go file or directory> ...
	Flags:
			-ignore REGEX
	```

	Examples

	```bash
	$ gocognit "_test|vendor/|code_gen|" .
	```

- Ignoring specific functions 

	```go
	//gocognit:ignore
	func f1() {
		// ...
	}
		
	//gocognit:ignore
	var f2 = func() {
		// ...
	}
	```

**Comment by uudashr** [comment]
You are using `gocognit` command line?
Can you explain why you want to ignore?

**Comment by krasi-georgiev** [comment]
for example the main function is something that can contain a lot of logic that can't be broken down and it is ok to have more complexity

for example having run groups is something that usually results in larger main files

**Comment by uudashr** [comment]
Understand, we can add this feature of course.

Btw , I want to understand your needs.
`gocognit` can be used directly using the command line interface or we can use the API as well.
For example, if we prefer to use linter runners such as [golangci-lint](https://github.com/golangci/golangci-lint), then we can use ignore feature provided by golangci-lint.

For your case, are you using `gocognit` CLI?

**Comment by uudashr** [comment]
Would love to have @wy98926 in the discussion (since the original request is from @wy98926 )

**Comment by ainar-g** [comment]
I am not OP, but to add to the discussion, I'm using `gocognit` CLI, and I have generated code (protobuf) I'd like to ignore.  Either adding `--ignore`, like `gocyclo`, or just ignoring all generated files by default would do.

**Comment by Nr18** [comment]
Just ran into this issue as well, I use the cli tool as followed:

```shell
gocognit -over 3 -avg .
```

I also use `gocyclo` in the following matter:

```shell
gocyclo -over 3 -avg -ignore "_test" .
```

Would be great to have the same option as `gocyclo` to exclude test files. Typically you want to keep your methods in your application code small and simple. (Hence using this complexity tool to test this)

But unit tests are way more verbose and can have multiple assertions. Resulting in a higher cognitive complexity score.

Not sure if my use case is the same as @wy98926, but at first glance it does look like it. It was that this issue already existed otherwise I would have opened it myself.

**Comment by uudashr** [comment]
Implemented in #27 
The `gocognit:ignore` directive extracted on #29

**Comment by Nr18** [comment]
Thankd @uudashr
